### DISCUSSION TITLE
How to clear the file history in the dock menu?

### DISCUSSION THREAD

**Discussion by imraax** [zed-industries/zed#16339] [Feature Requests]
<img width="264" alt="截屏2024-08-16 13 50 27" src="https://github.com/user-attachments/assets/2d9e15bd-e74f-4567-9485-0ebe6778b92a">

How to clear the file history in the dock menu?

**Comment by SomeoneToIgnore** [comment]
There's no way for that yet.
How other editors allow doing that, e.g. VSCode?

**Reply by imraax** [reply]
Hmm...the text editor of the macos clears it on the menu "File -> Recent -> Clear". I tried to find the reason, and then I saw `LSSharedFileList`  file in some articles, I found it  on `~/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments` and deleted it, but it didn't change anything.

**Reply by SomeoneToIgnore** [reply]
Oh, I see.

The whole story is somewhat perplexed due to Zed's open recent having only directories, ergo we never had a sub menu there.
Makes sense to add it for sure, along with the cleanup button — and most of the work seems related to Zed's functionality beyond calling a proper macOS platform method for clearing.

**Reply by imraax** [reply]
The interesting thing is that when I open a new file on zed and the file appears on dock menu, the `dev.zed.zed.sfl3` file is regenerated.

**Reply by lict99** [reply]
1. Quit Zed
2. Delete `~/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments/dev.zed.zed.sfl3`
3. Open Terminal and run `killall sharedfilelistd` (**Don't forget this!**)
4. Reopen Zed

Thanks [Hide/remove list of recent files on right click on dock icon MAC](https://forum.sublimetext.com/t/hide-remove-list-of-recent-files-on-right-click-on-dock-icon-mac/65920/8)

**Reply by aybarsnazlica** [reply]
In my case the file name to delete was dev.zed.zed.sfl4.

**Comment by lict99** [comment]
1. Quit Zed
2. Delete `~/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments/dev.zed.zed.sfl3`
3. Open Terminal and run `killall sharedfilelistd` (**Don't forget this!**)
4. Reopen Zed

Thanks [Hide/remove list of recent files on right click on dock icon MAC](https://forum.sublimetext.com/t/hide-remove-list-of-recent-files-on-right-click-on-dock-icon-mac/65920/8)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
